src/main/scala/consul/v1/common/ConsulRequestBasics.scala

```diff
@@ -36,15 +39,18 @@ object ConsulRequestBasics {
     }
   }
 
-  private def genRequestMaker[A,B](path: String, httpFunc: WSRequest => Future[WSResponse])(responseTransformer: WSResponse => B)(body: B => A)(implicit executionContext: ExecutionContext): Future[A] = {
-
-    Try(httpFunc(WS.url(path))) match {
+  private def genRequestMaker[A,B](path: String, httpFunc: HttpFunc)(responseTransformer: WSResponse => B)(body: B => A)(implicit executionContext: ExecutionContext): Future[A] = {
+    Try((withToken(token) andThen httpFunc)(WS.url(path))) match {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/Codacy/scala-consul/file/1255759080/issues/source?bid=2788492&fileBranchId=2939115#l43)